SafeLoggingHttpClientConnectionManager records where the pool is closed

carterkozak opened the pull request:
==COMMIT_MSG==
SafeLoggingHttpClientConnectionManager records where the pool is closed
==COMMIT_MSG==

carterkozak committed:
SafeLoggingHttpClientConnectionManager records where the pool is closed

carterkozak committed:
Add generated changelog entries

iamdanfox commented on dialogue-core/src/main/java/com/palantir/dialogue/core/PinUntilErrorNodeSelectionStrategyChannel.java:
```diff
@@ -182,6 +191,19 @@ private OptionalInt incrementHostIfNecessary(int pin) {
+        @Value.Derived
+        default CoarseExponentialDecayReservoir failureReservoir() {
+            return new CoarseExponentialDecayReservoir(clock()::read, Duration.ofSeconds(10));
+        }
```
think this stuff was a intended for a separate PR right?

carterkozak replied:
doh

carterkozak committed:
Might as well be threadsafe

The racy volatile read/set wouldn't be problematic, but there's
no harm in making this completely safe.

carterkozak commented:
:robot: :popcorn:

svc-autorelease commented:
Released [1.33.0](https://github.com/palantir/dialogue/releases/tag/1.33.0)
